pkg/autoscaler/metrics/collector_test.go

```diff
@@ -116,13 +117,92 @@ func TestMetricCollectorCRUD(t *testing.T) {
 	})
 }
 
+func TestMetricCollectorScraperMovingTime(t *testing.T) {
+	logger := TestLogger(t)
+
+	mtp := &fake.ManualTickProvider{
+		Channel: make(chan time.Time),
+	}
+	now := time.Now()
+	fc := fake.Clock{
+		FakeClock: clock.NewFakeClock(now),
+		TP:        mtp,
+	}
+	metricKey := types.NamespacedName{Namespace: defaultNamespace, Name: defaultName}
+	const (
+		reportConcurrency = 10
+		reportRPS         = 20
+		wantConcurrency   = 7 * 10 / 2
+		wantRPS           = 7 * 20 / 2
+		wantPConcurrency  = 7 * 10 / 2
+		wantPRPS          = 7 * 20 / 2
+	)
+	stat := Stat{
+		PodName:                   "testPod",
+		AverageConcurrentRequests: reportConcurrency,
+		RequestCount:              reportRPS,
+	}
+	scraper := &testScraper{
+		s: func() (Stat, error) {
+			return stat, nil
+		},
+	}
+	factory := scraperFactory(scraper, nil)
+
+	coll := NewMetricCollector(factory, logger)
+	coll.clock = fc
+	coll.CreateOrUpdate(&defaultMetric)
+
+	// Tick three times.  Time doesn't matter since we use the time on the Stat.
+	for i := 0; i < 3; i++ {
+		mtp.Channel <- now
+	}
+	now = now.Add(time.Second)
+	fc.SetTime(now)
+	for i := 0; i < 4; i++ {
+		mtp.Channel <- now
+	}
+	var gotRPS, gotConcurrency, panicRPS, panicConcurrency float64
+	// Poll to see that the async loop completed.
+	wait.PollImmediate(10*time.Millisecond, 2*time.Second, func() (bool, error) {
```

Comment:
Looks like this poll is only to save some time if the condition meets before the 2 seconds timeout, right?

Reply:
Not sure what you mean? The collector is an async process reading from channel. So we poll to see when this is executed.
it's the same in all the other tests.

Reply:
I mean excepting for saving time, is there any other benefits to use poll instead of sleep.

Reply:
Well poll is cool, sleep is so 80ies.